db.go

```diff
@@ -63,3 +64,28 @@ func runMigrations(dsn string) error {
 	}
 	return err
 }
+
+// DropOldPartition drops number of days old partitions of pmm.metrics in ClickHouse.
+func DropOldPartition(db *sqlx.DB, days uint) {
+	partitions := []string{}
+	const query = `
+SELECT
+DISTINCT partition
+FROM system.parts
+WHERE
+toDate(replaceRegexpOne(partition, '^(\\d\\d\\d\\d)(\\d\\d)(\\d\\d)$', '\\1-\\2-\\3')) < toDate(now() - toIntervalDay(?))
```

Comment:
Can we use `toYYYYMMDD` as in table definition instead of regexp?

We also should add ORDER BY to deleted older partitions first.

Reply:
@AlekSi, good point. I tried such variant as well.
But found, comparing two toYYYYMMDD is not the best solution as `partition` stores as a `String` in table.
So it better to parse string as date and compare two integers.

```
ClickHouse client version 19.5.4.22 (official build).
Connecting to localhost:9000 as user default.
Connected to ClickHouse server version 19.5.4 revision 54417.

0182ae3df148 :) SELECT DISTINCT partition FROM system.parts;

SELECT DISTINCT partition
FROM system.parts 

┌─partition─┐
│ 20190101  │
│ 20190102  │
└───────────┘

2 rows in set. Elapsed: 0.004 sec. 

0182ae3df148 :) SELECT DISTINCT partition FROM system.parts WHERE partition < toYYYYMMDD(now())

SELECT DISTINCT partition
FROM system.parts 
WHERE partition < toYYYYMMDD(now())

Received exception from server (version 19.5.4):
Code: 43. DB::Exception: Received from localhost:9000, 127.0.0.1. DB::Exception: Illegal types of arguments (String, UInt32) of function less. 

0 rows in set. Elapsed: 0.168 sec. 

0182ae3df148 :) desc system.parts  \G

DESCRIBE TABLE system.parts

Row 1:
──────
name:               partition
type:               String
default_type:       
default_expression: 
comment:            
codec_expression:   

```

Reply:
@AlekSi , thanks -it really simplifies query.